models/user_avatar.go

```diff
@@ -39,10 +39,9 @@ func (u *User) generateRandomAvatar(e Engine) error {
 		return fmt.Errorf("RandomImage: %v", err)
 	}
 
-	if u.Avatar == "" {
-		u.Avatar = HashEmail(u.AvatarEmail)
-	}
+	u.Avatar = HashEmail(seed)
```

Comment:
So you definitely intend the avatar always be overwritten if you call `generateRandomAvatar`?

Reply:
isnt this func only exec if no avatar afailable?

so it should be fine ...